std/machines/arith17.asm

```diff
@@ -12,19 +12,23 @@ use std::prelude::Query;
 use std::machines::range::Byte;
 use std::machines::range::Byte2;
 
-// Arithmetic machine, inspired by Polygon's 256-Bit Arith machine: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
-machine Arith16(byte: Byte, byte2: Byte2) with
+// Implements 32-Bit multiplication and division.
+// Inspired by Polygon's 256-Bit Arith machine: https://github.com/0xPolygonHermez/zkevm-proverjs/blob/main/pil/arith.pil
+// Requires the field to contain at least 17 bits.
+machine Arith17(byte: Byte, byte2: Byte2) with
     latch: CLK8_7,
     operation_id: is_division,
     // Allow this machine to be connected via a permutation
     call_selectors: sel,
 {
+    std::check::assert(std::field::modulus() >= 2**18, || "Field too small.");
```

Comment:
```suggestion
    std::check::assert(std::field::modulus() >= 2**19, || "Field too small.");
```

I think a 18-Bit field would overflow when computing `0xffffff * 0xffffff + 0xffffff`:
- The maximal limb is $2^8 - 1$
- The maximum product of two limbs is $(2^8 - 1)^2 = 2^{16} - 2 \cdot 2^8 + 1$
- To compute `x1 * y1 + x2`, we add up to 4 products and add one extra byte, so the maximum value would be:
  $4 * (2^{16} - 2 \cdot 2^8 + 1) + 2^8 - 1$
  $= 2^{18} - 2^{11} +2^8 + 3$

Reply:
bad gpt

Reply:
Wait, that's below $2^18$... But we're also adding a carry, which is currently constrained to be 16-Bit, so then it could overflow. Which I think is not necessary, we could have a smaller carry and stay within 18 bits?

So I'd say this current bound is wrong, but we could make it work with that bound.

Reply:
I'll just up the bound. The important thing is to stay under BB M31 KB

Reply:
done